rewrite-maven/src/main/java/org/openrewrite/maven/MavenSettings.java

```diff
@@ -83,6 +84,29 @@ public MavenSettings(@Nullable String localRepository, @Nullable Profiles profil
         this.servers = servers;
     }
 
+    public List<Profile> activeProfiles(final Iterable<String> userSpecifiedProfiles) {
```

Comment:
I made the code a little more production-ready.  I did copy this, due to my ignorance with lombok, between the model meant for POM files and the model meant for Maven settings filesI need to add tests for the changes that deal with Maven settings.xml profiles, but am hoping to get a bit of feedback on this approach.

Reply:
@timtebeek I pushed one way to isolate code in [f9bea1a](https://github.com/openrewrite/rewrite/pull/4270/commits/f9bea1a363414cbf8a6cbe9ec0f400a6167c86a9).  I'm far from happy with it.  I'll gladly remove/revert that commit if you are OK with copying this `activeProfiles` algorithm in 2 places.  Just let me know.